File: src/ser/scripts/my_client.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import sys
import rospy
from ser.srv import my_srv



def calculate_client(a,b):
    rospy.wait_for_service("my_service")
    try:
        my_service_client = rospy.ServiceProxy("my_service", my_srv)
        res = my_service_client(a,b)
        return res.result
    except rospy.ServiceException as e:
        rospy.logerr("Service call faild : %s", e)
        
if __name__ == "__main__":
    rospy.init_node("my_client")
    if len(sys.argv) != 3:
        rospy.loginfo("usage : rosrun ser my_client.py a, b")
        rospy.loginfo("a, b : int64")
        sys.exit()
    else:
        a = int(sys.argv[1])
        b = int(sys.argv[2])
        res = calculate_client(a,b)
        rospy.loginfo("send: a=%d, b=%d",a,b)
        rospy.loginfo("receive: result=%d",res)

File: src/ser/src/my_client.cpp
#include "ros/ros.h"
#include "ser/my_srv.h"
#include <cstdlib>

int main(int argc, char** argv)
{
    ros::init(argc,argv, "my_client");

    if(argc != 3)
    {
        ROS_INFO("cmd: rosrun ser my_client arg0 arg1");
        ROS_INFO("arg1, arg2 : int64 number");

        return 1;
    }

    ros::NodeHandle nh;
    ros::ServiceClient my_service_client = nh.serviceClient<ser::my_srv>("my_service");

    ser::my_srv srv;
    srv.request.a = atoll(argv[1]);
    srv.request.b = atoll(argv[2]);

    if(my_service_client.call(srv))
    {
        ROS_INFO("send srv a, b: %ld + %ld",srv.request.a, srv.request.b);
        ROS_INFO("receive srv result : %ld",srv.response.result);
    }
    else
    {
        ROS_ERROR("Fail to Call service");
        return 1;
    }

    return 0;
}

